internal/ingress/controller/parser/parser.go

```diff
@@ -125,6 +125,7 @@ var supportedCreds = sets.NewString(
 )
 
 var validProtocols = regexp.MustCompile(`\Ahttps$|\Ahttp$|\Agrpc$|\Agrpcs|\Atcp|\Atls$`)
+var validMethods = regexp.MustCompile(`\A[A-Z]+$`)
```

Comment:
This needs to be stricter.
A wrong annotation can block all updates from progressing and hence having strict validation is very important.
Please change the regex to accept only valid HTTP method names that kong accepts.

Reply:
Seems like Kong accepts `FOO` as a valid HTTP method.
Although custom HTTP method are a thing, let's be strict here and accept only the standardized methods.

Reply:
Per meeting discussion: yeah, it's weird, but it's technically a valid configuration and unlikely to cause confusion in practice (it doesn't appear to for non-controller managed instances, which allow the same range of input). Leaving as-is.